app/frontend/src/components/designer/FormViewer.vue

```diff
@@ -100,6 +102,14 @@ export default {
           }
           this.form = response.data;
           this.formSchema = response.data.schema;
+        } else if (this.draftId) {
+          // If getting for a specific draft version of the form for preview
+          response = await formService.readDraft(this.formId, this.draftId);
+          if (!response.data || !response.data.schema) {
+            throw new Error(`No schema in response. DraftId: ${this.draftId}`);
```

Comment:
Should consider invoking the notification popup for this.

Reply:
The catch handler that the error would get thrown to (130) has the notification handler.